### PR TITLE
Adding enum for debug parameters

### PR DESCRIPTION
**Pull request by joesus**
## Checklist

- [x] I've read the [Contributing Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've completed the [Contributor License Agreement](https://developers.facebook.com/opensource/cla)
- [x] I've ensured that all existing tests pass and added tests (when/where necessary)
- [x] I've ensured that my code lints properly: (`swiftlint && swiftlint autocorrect --format`)
- [x] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [x] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

## Pull Request Details

Adds simple enum to contain debug parameters

## Test Plan

Test Plan: Tests should pass, project should build.

### CODE REVIEW

File: Sources/Core/FacebookCoreTests/GraphApiDebugParameterTests.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    [GraphApiDebugParameter.info, .warning].forEach { parameter in
+      switch parameter {
+      case .info, .warning:
+        break
+      }
+    }
+  }
+
+  func testRawValues() {
+    let values = [GraphApiDebugParameter.info, .warning]
```
Comment (codytwinton) [inline Sources/Core/FacebookCoreTests/GraphApiDebugParameterTests.swift]:
```suggestion
    let values: [GraphApiDebugParameter] = [.info, .warning]
```

Explicit typing can even, at times, improve build times.

File: Sources/Core/GraphApiDebugParameter.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+enum GraphApiDebugParameter: String {
```
Comment (codytwinton) [inline Sources/Core/GraphApiDebugParameter.swift]:
```suggestion
enum GraphApiDebugParameter: String, CaseIterable {
```

File: Sources/Core/GraphApiDebugParameter.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  case info, warning
```
Comment (codytwinton) [inline Sources/Core/GraphApiDebugParameter.swift]:
```suggestion
  case info
  case warning
```

Better for blame

File: Sources/Core/FacebookCoreTests/GraphApiDebugParameterTests.swift [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+// swiftlint:disable multiline_arguments
+
+@testable import FacebookCore
+import XCTest
+
+class GraphApiDebugParameterTests: XCTestCase {
+
+  func testKnownValues() {
+    [GraphApiDebugParameter.info, .warning].forEach { parameter in
```
Comment (codytwinton) [inline Sources/Core/FacebookCoreTests/GraphApiDebugParameterTests.swift]:
```suggestion
    let expected: [GraphApiDebugParameter] = [.info, .warning]
    XCTAssertEqual(expected, GraphApiDebugParameter.allCases, "Missing GraphApiDebugParameter")
    expected.forEach { parameter in
```

File: Sources/Core/FacebookCoreTests/GraphApiDebugParameterTests.swift (hunk 1 above)
Comment (joesus) [reply Sources/Core/FacebookCoreTests/GraphApiDebugParameterTests.swift]: updating to use CaseIterable - allCases

File: Sources/FacebookCoreTests/GraphApiDebugParameterTests.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+// swiftlint:disable multiline_arguments
+
+@testable import FacebookCore
+import XCTest
+
+class GraphApiDebugParameterTests: XCTestCase {
+
+  func testKnownValues() {
+    [GraphApiDebugParameter.info, .warning].forEach { parameter in
```
Comment (codytwinton) [inline Sources/FacebookCoreTests/GraphApiDebugParameterTests.swift]: Change to compare against `allCases`

File: Sources/FacebookCoreTests/GraphApiDebugParameterTests.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// swiftlint:disable multiline_arguments
+
+@testable import FacebookCore
+import XCTest
+
+class GraphApiDebugParameterTests: XCTestCase {
+
```
Comment (robtimp) [inline Sources/FacebookCoreTests/GraphApiDebugParameterTests.swift]: These tests feel to me like they’re testing Apple’s code much more than our own (i.e. check if `allCases` and `rawValues` for enums work as they’re supposed to). Thoughts?

Comment (codytwinton) [reply Sources/FacebookCoreTests/GraphApiDebugParameterTests.swift]: Disagree. Both are beneficial to ensure that we're covering all cases should the enum ever change. Super easy to mass-rename all instances in code using Xcode, but the testing against rawValues ensures that we change with full disclosure.

Comment (joesus) [reply Sources/FacebookCoreTests/GraphApiDebugParameterTests.swift]: I also am a little bit trying to encourage the idea that if something needs changing it's a conscious decision and not just an easy hack. Sort of the guardrails approach.

**Pull request merged**